src/Sulu/Bundle/PreviewBundle/Preview/Renderer/PreviewRenderer.php

```diff
@@ -155,7 +155,7 @@ public function render($object, $id, $webspaceKey, $locale, $partial = false)
         $this->eventDispatcher->dispatch(Events::PRE_RENDER, new PreRenderEvent($attributes));
 
         try {
-            $response = $this->httpKernel->handle($request, HttpKernelInterface::SUB_REQUEST, false);
+            $response = $this->httpKernel->handle($request, HttpKernelInterface::MASTER_REQUEST, false);
```

Comment:
If you'd like to make sure it works for both kind of request you would also have to test both variants, not change the existing one 😉